server/hotrod/src/main/scala/org/infinispan/server/hotrod/CacheDecodeContext.scala

```diff
@@ -261,18 +260,28 @@ class CacheDecodeContext(server: HotRodServer) extends ServerConstants with Log
     * Otherwise it's just considered number of seconds from
     * now and it's returned in milliseconds unit.
     */
-   protected def toMillis(lifespan: Int): Long = {
-      if (lifespan > SecondsInAMonth) {
-         val unixTimeExpiry = TimeUnit.SECONDS.toMillis(lifespan) - System.currentTimeMillis
+   protected def toMillis(param: ExpirationParam): Long = {
```

Comment:
ok, removing 30d check for 2.2 and keeping the behaviour for < 2.2 is not complicated and worst case 2.2 clients relying on 30d behaviour will just need to call cache ops sending the right unit, now that we support all of them